### PR TITLE
[Fix] Fix docs:1_exist_data_model.md,how_to.md,useful_tools.md,changelog_v2.x.md,api.rst

### PR DESCRIPTION
**Pull request by wanghonglie**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification
update: 1_exist_data_model.md,how_to.md,useful_tools.md,api.rst
rename: changelog.md,changelog_v2.x.md

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docs/en/1_exist_data_model.md [hunk 1]
```diff
@@ -10,36 +10,78 @@ MMDetection provides hundreds of existing and existing detection models in [Mode
 
 By inference, we mean using trained models to detect objects on images. In MMDetection, a model is defined by a configuration file and existing model parameters are save in a checkpoint file.
 
-To start with, we recommend [Faster RCNN](https://github.com/open-mmlab/mmdetection/tree/master/configs/faster_rcnn) with this [configuration file](https://github.com/open-mmlab/mmdetection/blob/master/configs/faster_rcnn/faster_rcnn_r50_fpn_1x_coco.py) and this [checkpoint file](https://download.openmmlab.com/mmdetection/v2.0/faster_rcnn/faster_rcnn_r50_fpn_1x_coco/faster_rcnn_r50_fpn_1x_coco_20200130-047c8118.pth). It is recommended to download the checkpoint file to `checkpoints` directory.
+To start with, we recommend [Faster RCNN](https://github.com/open-mmlab/mmdetection/tree/master/configs/faster_rcnn) with this [configuration file](https://github.com/open-mmlab/mmdetection/blob/master/configs/faster_rcnn/faster-rcnn_r50_fpn_1x_coco.py) and this [checkpoint file](https://download.openmmlab.com/mmdetection/v2.0/faster_rcnn/faster_rcnn_r50_fpn_1x_coco/faster_rcnn_r50_fpn_1x_coco_20200130-047c8118.pth). It is recommended to download the checkpoint file to `checkpoints` directory.
```
Comment (BIGWangYuDong) [inline docs/en/1_exist_data_model.md:13]: maybe we should change `master` -> `dev-3.x` ?

File: docs/en/useful_tools.md [hunk 2]
```diff
@@ -146,18 +146,21 @@ python tools/analysis_tools/coco_error_analysis.py ${RESULT} ${OUT_DIR} [-h] [--
...
 python tools/test.py \
-       configs/mask_rcnn/mask_rcnn_r50_fpn_1x_coco.py \
+       configs/mask_rcnn/mask-rcnn_r50_fpn_1x_coco.py \
        checkpoint/mask_rcnn_r50_fpn_1x_coco_20200205-d4b0c5d6.pth \
-       --format-only \
-       --options "jsonfile_prefix=./results"
 ```
 
-1. Get COCO bbox error results per category , save analyze result images to the directory `results/`
+1. Get COCO bbox error results per category , save analyze result images to the directory(In  [config](https://github.com/wanghonglie/mmdetection/tree/master/configs/_base_/datasets/coco_instance.py) the default directory is './work_dirs/coco_instance/test')
```
Comment (BIGWangYuDong) [inline docs/en/useful_tools.md:163]:
```suggestion
1. Get COCO bbox error results per category , save analyze result images to the directory(In [config](https://github.com/wanghonglie/mmdetection/tree/master/configs/_base_/datasets/coco_instance.py) the default directory is './work_dirs/coco_instance/test')
```

File: docs/en/1_exist_data_model.md [hunk 3]
```diff
@@ -10,36 +10,80 @@ MMDetection provides hundreds of existing and existing detection models in [Mode
...
+# the dataset_meta is loaded from the checkpoint and
+# then pass to the model in init_detector
+visualizer.dataset_meta = model.dataset_meta
+
+# The interval of show (s), 0 is block
+wait_time = 1
+
+video_reader = mmcv.VideoReader('video.mp4')
+
+for frame in mmcv.track_iter_progress(video_reader):
```
Comment (chhluo) [inline docs/en/1_exist_data_model.md:74]: track_iter_progress is moved to mmengine.utils

File: docs/en/1_exist_data_model.md [hunk 4]
```diff
@@ -340,73 +379,70 @@ Assuming that you have already downloaded the checkpoints to the directory `chec
...
        --out results.pkl \
-       --eval bbox segm \
-       --options "classwise=True"
+       --cfg-options test_evaluator.classwise=True
    ```
 
 6. Test Mask R-CNN on COCO test-dev with 8 GPUs, and generate JSON files for submitting to the official evaluation server.
    Config and checkpoint files are available [here](https://github.com/open-mmlab/mmdetection/tree/master/configs/mask_rcnn).
 
+   Replace the original test_evaluator and test_dataloader with test_evaluator and test_dataloader in the comment in [config](https://github.com/wanghonglie/mmdetection/tree/master/configs/_base_/datasets/coco_instance.py) and run:
```
Comment (chhluo) [inline docs/en/1_exist_data_model.md:422]: -> https://github.com/open-mmlab/mmdetection/tree/dev-3.x/configs/_base_/datasets/coco_instance.py

File: docs/en/1_exist_data_model.md [hunk 5]
```diff
@@ -340,73 +379,70 @@ Assuming that you have already downloaded the checkpoints to the directory `chec
...
-       --options "jsonfile_prefix=./mask_rcnn_test-dev_results"
    ```
 
-   This command generates two JSON files `mask_rcnn_test-dev_results.bbox.json` and `mask_rcnn_test-dev_results.segm.json`.
+   This command generates two JSON files `./work_dirs/coco_instance/test.bbox.json` and `./work_dirs/coco_instance/test.segm.json`.
 
 7. Test Mask R-CNN on Cityscapes test with 8 GPUs, and generate txt and png files for submitting to the official evaluation server.
    Config and checkpoint files are available [here](https://github.com/open-mmlab/mmdetection/tree/master/configs/cityscapes).
 
+   Replace the original test_evaluator and test_dataloader with test_evaluator and test_dataloader in the comment in [config](https://github.com/wanghonglie/mmdetection/tree/master/configs/_base_/datasets/cityscapes_instance.py) and run:
```
Comment (chhluo) [inline docs/en/1_exist_data_model.md:436]: -> https://github.com/open-mmlab/mmdetection/tree/dev-3.x/configs/_base_/datasets/cityscapes_instance.py

File: docs/en/1_exist_data_model.md [hunk 6]
```diff
@@ -427,15 +463,13 @@ arguments：
 - `OUT`: The output annotation json file name. The save dir is in the same directory as `IMG_PATH`.
 - `exclude-extensions`: The suffix of images to be excluded, such as 'png' and 'bmp'.
 
-After the conversion is complete, you can use the following command to test
+After the conversion is complete, you need to replace the original test_evaluator and test_dataloader with test_evaluator and test_dataloader in the comment in [config](https://github.com/wanghonglie/mmdetection/tree/master/configs/_base_/datasets/coco_detection.py)(find which dataset in 'configs/_base_/datasets' the current config corresponds to) and run:
```
Comment (chhluo) [inline docs/en/1_exist_data_model.md:466]: be careful with the link

File: docs/en/tutorials/how_to.md [hunk 7]
```diff
@@ -39,7 +39,7 @@ model = dict(
 MMClassification also provides a wrapper for the PyTorch Image Models (timm) backbone network, users can directly use the backbone network in timm through MMClassification. Suppose you want to use EfficientNet-B1 as the backbone network of RetinaNet, the example config is as the following.
 
 ```python
-# https://github.com/open-mmlab/mmdetection/blob/master/configs/timm_example/retinanet_timm_efficientnet_b1_fpn_1x_coco.py
+# https://github.com/open-mmlab/mmdetection/blob/master/configs/timm_example/retinanet_timm-efficientnet-b1_fpn_1x_coco.py
```
Comment (chhluo) [inline docs/en/tutorials/how_to.md:42]: master -> dev-3.x

File: docs/en/useful_tools.md (hunk 2 above)
Comment (chhluo) [reply docs/en/useful_tools.md:163]: wanghonglie -> openmmlab, master -> dev-3.x

File: docs/en/useful_tools.md [hunk 8]
```diff
@@ -305,15 +308,9 @@ comparisons, but double check it before you adopt it in technical reports or pap
...
-
-We provide a script to convert model to [ONNX](https://github.com/onnx/onnx) format. We also support comparing the output results between Pytorch and ONNX model for verification.
-
-```shell
-python tools/deployment/pytorch2onnx.py ${CONFIG_FILE} ${CHECKPOINT_FILE} --output-file ${ONNX_FILE} [--shape ${INPUT_SHAPE} --verify]
-```
+### MMDetection model to ONNX
 
-**Note**: This tool is still experimental. Some customized operators are not supported for now. For a detailed description of the usage and the list of supported models, please refer to [pytorch2onnx](tutorials/pytorch2onnx.md).
+We provide a script to convert model to [ONNX](https://github.com/onnx/onnx) format. We also support comparing the output results between Pytorch and ONNX model for verification. More details can refer to [mmdeploy](https://github.com/open-mmlab/mmdeployhttps://github.com/open-mmlab/mmdeploy)
```
Comment (chhluo) [inline docs/en/useful_tools.md:313]: https://github.com/open-mmlab/mmdeployhttps://github.com/open-mmlab/mmdeploy ?

File: docs/en/1_exist_data_model.md [hunk 9]
```diff
@@ -306,20 +348,17 @@ bash tools/dist_test.sh \
...
-- `EVAL_METRICS`: Items to be evaluated on the results. Allowed values depend on the dataset, e.g., `proposal_fast`, `proposal`, `bbox`, `segm` are available for COCO, `mAP`, `recall` for PASCAL VOC. Cityscapes could be evaluated by `cityscapes` as well as all COCO metrics.
 - `--show`: If specified, detection results will be plotted on the images and shown in a new window. It is only applicable to single GPU testing and used for debugging and visualization. Please make sure that GUI is available in your environment. Otherwise, you may encounter an error like `cannot connect to X server`.
 - `--show-dir`: If specified, detection results will be plotted on the images and saved to the specified directory. It is only applicable to single GPU testing and used for debugging and visualization. You do NOT need a GUI available in your environment for using this option.
-- `--show-score-thr`: If specified, detections with scores below this threshold will be removed.
+- `--work-dir`: If specified, detection results of containing evaluation metrics will be saved to the specified directory.
```
Comment (zytx121) [inline docs/en/1_exist_data_model.md:360]:
```suggestion
- `--work-dir`: If specified, detection results containing evaluation metrics will be saved to the specified directory.
```

File: docs/en/tutorials/how_to.md [hunk 10]
```diff
@@ -65,14 +65,14 @@ optimizer = dict(type='SGD', lr=0.01, momentum=0.9, weight_decay=0.0001)
 
 `type='mmcls.TIMMBackbone'` means use the `TIMMBackbone` class from MMClassification in MMDetection, and the model used is `EfficientNet-B1`, where `mmcls` means the MMClassification repo and `TIMMBackbone` means the TIMMBackbone wrapper implemented in MMClassification.
 
-For the principle of the Hierarchy Registry, please refer to the [MMCV document](https://github.com/open-mmlab/mmcv/blob/master/docs/en/understand_mmcv/registry.md#hierarchy-registry). For how to use other backbones in MMClassification, you can refer to the [MMClassification document](https://github.com/open-mmlab/mmclassification/blob/master/docs/en/tutorials/config.md).
+For the principle of the Hierarchy Registry, please refer to the [MMCV document](https://github.com/open-mmlab/mmcv/blob/dev-3.x/docs/en/understand_mmcv/registry.md#hierarchy-registry). For how to use other backbones in MMClassification, you can refer to the [MMClassification document](https://github.com/open-mmlab/mmclassification/blob/dev-3.x/docs/en/tutorials/config.md).
```
Comment (chhluo) [inline docs/en/tutorials/how_to.md:68]: 3.x is only for mmdet, 2.x or1.x for other repos.

File: docs/en/api.rst [hunk 11]
```diff
@@ -3,105 +3,152 @@ mmdet.apis
...
 
-anchor
+datasets
 ^^^^^^^^^^
-.. automodule:: mmdet.core.anchor
+.. automodule:: mmdet.datasets
     :members:
 
-bbox
+pipelines
```
Comment (RangiLyu) [inline docs/en/api.rst:14]:
```suggestion
transforms
```

**Review by RangiLyu — APPROVED** [review]

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**